### ISSUE TITLE
add random fucntion

### ISSUE THREAD

**Issue by anion155** [stanhebben/MineTweaker3#330]
As I understand, there is no function returned pseudo-random int. Can somebody provide some?

**Comment by VariableVixen** [comment]
I'm just a bit curious, what for?

**Comment by jaredlll08** [comment]
actually yea, this is not a good idea, because then recipes will be different on client and server.

Lets say that on the server, a recipe is added that gives X of an item
on the client, a recipe is added that gives Y of an item.

It is just not a good idea at all.

**Comment by anion155** [comment]
Well, in my case I'm playing with Agrarian skies 2. And Hardcore Questing Mode mod has a hearts as lives. It has normal heart and rotten heart, and I want to change crafting recipes of normal heart, so they will give sometimes rotten heart. I don't think synchronization is an issue. Same seed to Random class on client and server.

**Comment by jaredlll08** [comment]
oh! that is what you want, yea not gna happen, I thought you wanted a random int type thing.

Unfortunately, you can't do what you want, especially in minetweaker. sorry

**Comment by anion155** [comment]
All I need is "random int type thing", cause diff between rotten and normal hearts is damage value.

**Comment by jaredlll08** [comment]
yes but all that will happen is:

Scripts load
recipe gets added using random int
recipe now gives Item X
craft 500 times
Only receive item X 500 times, because the recipe got set and cached.
restart game
Scripts load
recipe gets added using random int
recipe now gives Item Y
craft 500 times
Only receive item Y 500 times, because the recipe got set and cached.

It will never work how you want it to unfortunatly

**Comment by anion155** [comment]
You didn't understand. I need custom result on every craft, so I need to use callback ("Function recipe" as it called in wiki).
Something like this:

```
recipes.addShapeless(<mod:heart>.marked('heart'), [<mod:cool_ingredient>], function(out, in, craft) {
  return out.heart.withDamage(rand() / 2 + 5);
});
```

**Comment by jaredlll08** [comment]
Yes, I fully understand what you want. And I'm telling you it isn't possible.

**Comment by VariableVixen** [comment]
There _cannot_ be a random element in MineTweaker scripts. It will not happen. Ever. Assuming it actually is possible - and I think that I may know how to make it work, at least in general - it is still far too much work. It would require rewriting most, if not all, of the mod.

**Comment by jaredlll08** [comment]
Ok let me rephrase. This is totally possible, fairly simple to do imho
Simply create a new item, that item has 2 nbt tags, one for item X and one for Item Y, then on the update method, just set the function to that item.
Now the reason that this will not be happening in minetweaker is, We do not add items and do not plan on adding items.

If you have more ideas, tell them to me, and if it is worth it, I could maybe whip up a minetweaker addon that will do it, but as of right now, Minetweaker as a mod will not add new items/blocks/liquids to the game
